### PR TITLE
fix: hardening pdf processing by disabling scripting in pdf viewer

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

**Review by JanAckermann — APPROVED** [review]

File: js/workersrc.js [hunk 1]
```diff
@@ -23,14 +23,16 @@ function deferredViewerConfig() {
...
+		PDFViewerApplicationOptions.set('workerSrc', head.getAttribute('data-workersrc'));
 		PDFViewerApplicationOptions.set('locale', getSanitizedCurrentLocale());
-		PDFViewerApplicationOptions.set('cMapUrl', document.getElementsByTagName('head')[0].getAttribute('data-cmapurl'));
-		PDFViewerApplicationOptions.set('sandboxBundleSrc', document.getElementsByTagName('head')[0].getAttribute('data-sandbox'));
+		PDFViewerApplicationOptions.set('cMapUrl', head.getAttribute('data-cmapurl'));
+		PDFViewerApplicationOptions.set('sandboxBundleSrc', head.getAttribute('data-sandbox'));
 		PDFViewerApplicationOptions.set('printResolution', 300);
 		PDFViewerApplicationOptions.set('externalLinkTarget', pdfjsLib.LinkTarget.BLANK);
 		PDFViewerApplicationOptions.set('isEvalSupported', false);
+		PDFViewerApplicationOptions.set('enableScripting', head.getAttribute('data-enableScriting'));
```
Comment (JanAckermann) [inline js/workersrc.js:35]: Do we need to cast the string ?

Comment (pako81) [reply js/workersrc.js:35]: `data-enableScriting`--> `data-enableScripting`

File: templates/viewer.php [hunk 2]
```diff
@@ -30,6 +34,7 @@
 <head data-workersrc="<?php p($urlGenerator->linkTo('files_pdfviewer', 'js/vendor/pdfjs/build/pdf.worker.js')) ?>?v=<?php p($version) ?>"
 	  data-cmapurl="<?php p($urlGenerator->linkTo('files_pdfviewer', 'js/vendor/pdfjs/web/cmaps/'))?>"
 	  data-sandbox="<?php p($urlGenerator->linkTo('files_pdfviewer', 'js/vendor/pdfjs/build/pdf.sandbox.js'))?>"
+	  data-enableScriting="<?php p($enableScripting ? 'true' : 'false') ?>"
```
Comment (pako81) [inline templates/viewer.php:37]: `data-enableScriting`--> `data-enableScripting`

**Review by JanAckermann — CHANGES_REQUESTED** [review]
Fix typo, check if bool cast is required

File: js/workersrc.js (hunk 1 above)
Comment (DeepDiver1975) [reply js/workersrc.js]: :see_no_evil:

**Review by JanAckermann — APPROVED** [review]

**Review by pako81 — APPROVED** [review]

**Comment by empty00** [comment]
@DeepDiver1975 could you please replay the report in HackerOne

**Pull request merged**